aten/src/ATen/Dispatch.h

```diff
@@ -10,30 +10,59 @@
     return __VA_ARGS__();                          \
   }
 
+namespace detail {
+
+inline at::ScalarType scalar_type(at::ScalarType s) {
+  return s;
+}
+
+C10_DEPRECATED_MESSAGE("passing at::Type to an AT_DISPATCH macro is deprecated, " \
```

Comment:
i'm curious whether these empty functions will be optimized out, and if so will the deprecation warning still be emitted? although if that's the case, there's probably not an easy way around that.

Reply:
The warning is suppose to be emitted at compile-time, so a compiler should be able to emit them regardless of the optimization.